Makefile

```diff
@@ -1,14 +1,27 @@
-GOFLAGS :=
-IMAGE_REPOSITORY_NAME ?= openshift
+.PHONY: all
 
-build:
-	go build -mod=vendor $(GOFLAGS) .
-.PHONY: build
+# Include the library makefile
+include $(addprefix ./vendor/github.com/openshift/build-machinery-go/make/, \
+    golang.mk \
+    targets/openshift/deps-gomod.mk \
+    targets/openshift/images.mk \
+)
 
-images:
-	imagebuilder -f Dockerfile -t $(IMAGE_REPOSITORY_NAME)/oauth-proxy .
-.PHONY: images
+IMAGE_REGISTRY?=registry.svc.ci.openshift.org
+
+# This will call a macro called "build-image" which will generate image specific targets based on the parameters:
+# $0 - macro name
+# $1 - target name
+# $2 - image ref
+# $3 - Dockerfile path
+# $4 - context directory for image build
+# It will generate target "image-$(1)" for building the image and binding it as a prerequisite to target "images".
+$(call build-image,ocp-oauth-proxy,$(IMAGE_REGISTRY)/ocp/4.4:oauth-proxy,./Dockerfile.rhel7,.)
 
 clean:
 	$(RM) ./oauth-proxy
 .PHONY: clean
+
+GO_BUILD_PACKAGES := .
+# avoid the test/ directory only containing the e2e tests
+GO_TEST_PACKAGES :=./ ./api/... ./cookie/... ./providers/... ./util/...
```

Comment:
Where is the target to run local e2es?

Reply:
that's tbd in #190, currently no tests are being run. Well, they are, in Travis, they attempt to build origin 3.11 and fail horribly. This repo is a mess.

Reply:
I suppose I'd at least hook it as when setting up the makefile:
```
test-e2e: GO_TEST_PACKAGES :=./test/e2e/...
test-e2e: GO_TEST_FLAGS += -v
test-e2e: GO_TEST_FLAGS += -timeout 3h
test-e2e: GO_TEST_FLAGS += -p 1
test-e2e: test-unit
.PHONY: test-e2e
``

Reply:
sure, I have that in a successive PR #148, but I'll add it here, too

Reply:
ok, you linked this PR before so I wasn't sure